jupyter_book_to_htmlbook/file_processing.py

```diff
@@ -84,14 +84,17 @@ def process_chapter_single_file(toc_element):
         return None, None
 
     # apply appropriate data-type (best guess)
-    if ch_name.lower() in front_matter:
-        if ch_name.lower() in allowed_data_types:
-            chapter['data-type'] = ch_name.lower()  # type: ignore
+
+    ch_stub = re.sub('[^a-zA-Z]', '', ch_name)
```

Comment:
This stub lets us discover that 'preface' is, indeed, inside '00-beginning-preface-some-after-cruft.ipynb', and add the appropriate `data-type`.